src/SiteConfig/LoginFormAuthenticator.php

```diff
@@ -31,7 +31,7 @@ public function login(SiteConfig $siteConfig, ClientInterface $guzzle)
 
         $guzzle->post(
             $siteConfig->getLoginUri(),
-            ['body' => $postFields, 'allow_redirects' => true, 'verify' => false]
+            ['body' => $postFields, 'allow_redirects' => true, 'verify' => false] + ($siteConfig->hasHttpHeaders() ? ['headers' => $siteConfig->getHttpHeaders()] : []),
```

Comment:
for code readability I'd set a `$param` variable with base values, then use an explicit `if` statement to add the `headers` key